src/adapter/src/catalog/builtin.rs

```diff
@@ -4215,6 +4215,24 @@ WHERE false
     ",
 };
 
+pub const MZ_SHOW_SOURCES: BuiltinView = BuiltinView {
+    name: "mz_show_sources",
+    schema: MZ_INTERNAL_SCHEMA,
+    sql: "CREATE VIEW mz_internal.mz_show_sources
+AS SELECT sources.name, sources.type, sources.size, clusters.name as cluster, schema_id, cluster_id,
+FROM mz_catalog.mz_sources AS sources
+LEFT JOIN mz_catalog.mz_clusters AS clusters ON clusters.id = sources.cluster_id",
```

Comment:
Can you remind me, why are these `LEFT JOIN`s instead of just `JOIN`s?

Reply:
I believe the issue I had with the JOIN on mz_clusters was that I was using OUTER JOIN, so changing to JOIN should be fine.

I thought about changing this but I can't run sqllogictest locally atm so didn't want to make more changes I couldn't test.

Reply:
I think we should change this to `JOIN` then to match the materialized view case. I'll make sure to kick off the CI build which should hopefully let us know if something failed.